x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/apikey/BulkUpdateApiKeyRequest.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.security.action.apikey;
+
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.xpack.core.security.authz.RoleDescriptor;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+
+public final class BulkUpdateApiKeyRequest extends BaseUpdateApiKeyRequest {
+
+    public static BulkUpdateApiKeyRequest usingApiKeyIds(String... ids) {
+        return new BulkUpdateApiKeyRequest(Arrays.stream(ids).toList(), null, null);
+    }
+
+    private final List<String> ids;
```

Comment:
Any reason for using `List` instead of `Set` for this field given we are not goiong to check for duplication?

Reply:
I don't see a strong reason to use a set here.

Using a list to me signals that the ids provided will be processed in the order they are given, whereas a set signals no particular order. I have a slight preference for the former (order). Deterministic order can come when an end-user issues a request, checks for failures, makes some changes, and re-issues the request for the same ids. We could get deterministic order with sets too but the `Set` interface does not communicate that to someone simply looking at the class/API.

From a performance perspective, a set on the request model would only help in the case where there is a large number of duplicates, which is an unusual corner case that is probably not worth optimizing for. 

Likewise, for the response and response builder classes, we essentially just need a container to append to and have no requirement for fast membership checks or uniqueness (since the implementation is already robust to duplicates in other ways). 

It's a nice bonus that for the happy path where all ids get a successfully updated, the order of the ids in the request have the same order as those in the response, if we use an array/list.

In terms of consistency with other requests and responses, I predominantly see arrays and lists (e.g., `InvalidateApiKeyRequest` uses `String[]`, `GetProfilesResponse` uses `Profile[]`).